src/Refactoring-UI-Tests/ReExtractToTemporaryDriverTest.class.st

```diff
@@ -0,0 +1,56 @@
+Class {
+	#name : 'ReExtractToTemporaryDriverTest',
+	#superclass : 'ReDriverTest',
+	#instVars : [
+		'driver'
+	],
+	#category : 'Refactoring-UI-Tests-Driver',
+	#package : 'Refactoring-UI-Tests',
+	#tag : 'Driver'
+}
+
+{ #category : 'tests' }
+ReExtractToTemporaryDriverTest >> testExtractToTemporaryExpectChanges [
+
+	| driverChanges |
+	driver := ReExtractToTemporaryDriver new
+					 extract: (130 to: 138)
+					 to: 'variable'
+		          from: #noMoveDefinition
+		          in: RBClassDataForRefactoringTest.
+
+	self setUpDriver: driver.
+
+	driver runRefactoring.
+	driverChanges := driver refactoring changes.
+	
+	self assert: driverChanges changes size equals: 1.
+]
+
+{ #category : 'tests' }
+ReExtractToTemporaryDriverTest >> testFailureInvalidVariableName [
+
+    driver := ReExtractToTemporaryDriver new
+        extract: (51 to: 60)
+        to: 'collection'
+        from: #modifiesCollection
+        in: RBClassDataForRefactoringTest.
+
+    self setUpDriver: driver.
+    
+    self should: [ driver runRefactoring ] raise: Error.
```

Comment:
It would be nice if we would expect as specific error as possible. I'm guessing this is `RBRefactoringError`?